packages/devtools_app/lib/src/flutter/provider.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../auto_dispose.dart';
+import '../globals.dart';
+import '../logging/logging_controller.dart';
+
+/// Container for controllers that should outlive individual screens of the app.
+@immutable
+class ProviderData implements DisposableController {
+  const ProviderData({@required this.loggingController})
+      : assert(loggingController != null);
+
+  /// Builds the default providers for the app.
+  factory ProviderData.defaultProviders() {
+    return ProviderData(
+      loggingController: LoggingController(
+        onLogCountStatusChanged: (_) {},
+        // TODO(djshuckerow): Use a notifier pattern for the logging controller.
+        // That way, it is visible if it has listeners and invisible otherwise.
+        isVisible: () => true,
+      ),
+    );
+  }
+
+  final LoggingController loggingController;
+
+  @override
+  void dispose() {
+    loggingController.dispose();
+  }
+}
+
+/// Provider for controllers that should outlive individual screens of the app.
+///
+/// [Initializer] builds a [Provider] after it has a connection to the VM
+/// service and it has loaded [ensureInspectorDependencies].
+class Provider extends StatefulWidget {
+  const Provider({Key key, this.child, this.overrideProviders})
+      : super(key: key);
+
+  final ProviderData Function() overrideProviders;
```

Comment:
nit: can this method be `@visibleForTesting` ?

Reply:
Done.